### DISCUSSION TITLE
Possible to get decomposed transform from 2 sets of given points?

### DISCUSSION THREAD

**Discussion by lsps9150414** [chrvadala/transformation-matrix#92] [Q&A] [answered]
I'm working on gesture handling where users can scale, drag, and rotate a shape with 2 fingers.

The gesture system can let me know (a) where the initial touchpoints are and (b) where the current touchpoints are.
I need to apply a transform style to the shape being manipulated to reflect the manipulation.

Is it possible for this lib to output the translate, scale, and rotate parameters (i.e. the decomposed transform matrix) by giving it 2 sets of points, the init touch points `{ (x1, y1), (x2, y2) }` and the final touch points `{ (x1', y1'), (x2', y2') }`?

**Comment by chrvadala** [comment, marked as answer]
Update 23 Oct 2022
Feature released with transformation-matrix@2.14.0

----------------------

Hello @lsps9150414 ,
currently there isn't this feature, but I opened an issue and I'll try to work on it in the next weeks.
Are you still looking for a solution (I understand that you posted this message long time ago).  Have you, in the meantime, found some interesting suggestion for me?

You can track progresses here https://github.com/chrvadala/transformation-matrix/issues/94

**Reply by lsps9150414** [reply]
Hey @chrvadala thanks for getting back to me. Yeah, I eventually stiched together a solution with https://github.com/axelpale/nudged which basically does what I described in the OP.

**Reply by chrvadala** [reply]
Wow, thanks for sharing it. It looks a complete library, with a lot of interesting features.
On my part, recently I started working on these two new methods: `fromOneMovingPoint` and `fromTwoMovingPoints`. They should address the case, but they are still under testing.

Currently the source code is available on this branch https://github.com/chrvadala/transformation-matrix/tree/moving-points, but I plan to convert it to a PR in the following days. I'll link the PR here for tracking purpose.

Thanks @lsps9150414

**Reply by chrvadala** [reply]
Feature released with transformation-matrix@2.14.0
